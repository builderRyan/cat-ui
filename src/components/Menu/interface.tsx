import * as React from 'react'

export interface InlineSubMenuProps {
  height?: number | string;
}
