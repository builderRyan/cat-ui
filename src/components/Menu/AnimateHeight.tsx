import * as React from 'react';
import { PureComponent } from 'react';
import classNames from 'classnames';

export interface AnimateHeightProps {
  /** 类名 */
  className?: string;
  /** 样式 */
  style?: any;
  duration: number;
  easing: string;
  /** 高度 */
  height: string | number;
}

/**
 * AnimateHeight
 */
class AnimateHeight extends PureComponent<AnimateHeightProps, any> {
  constructor(props: AnimateHeightProps) {
    super(props);
    this.state = {
      overflow: 'visible'
    };
  }

  static defaultProps = {
    duration: 250,
    easing: 'ease'
  };

  render() {
    const {
      className,
      style = {},
      height,
      duration,
      easing,
      children
    } = this.props;
    const { overflow } = this.state;
    const classes = classNames('cat-animate', className);

    const userTransition = style.transition ? `${style.transition},` : '';
    const transitionString = `${userTransition} height ${duration}ms ${easing}`;
    const componentStyle = {
      ...style,
      height,
      overflow: overflow || style.overflow
    };

    return (
      <div className={classes} style={componentStyle}>
        {children}
      </div>
    );
  }
}

export default AnimateHeight;
