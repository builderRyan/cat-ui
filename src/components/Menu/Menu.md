Example

```jsx
import Icon from '../Icon'
class Example extends React.Component {
  constructor (props) {
    super(props)
    this.state = {
      activeKey4: 'income',
      activeKey5: 'setting',
      activeKey6: 'dashboard',
      activeKey7: 'setting'
    }
    this.onItem1Click = this.onItem1Click.bind(this)
    this.onMenu4Change = this.onMenu4Change.bind(this)
    this.onMenu5Change = this.onMenu5Change.bind(this)
    this.onMenu6Change = this.onMenu6Change.bind(this)
    this.onMenu7Change = this.onMenu7Change.bind(this)
  }

  onItem1Click (e) {
    console.log('Example:onItem1Click', e)
  }

  onMenu4Change (key) {
    this.setState({ activeKey4: key })
  }

  onMenu5Change (key) {
    this.setState({ activeKey5: key })
  }

  onMenu6Change (key) {
    this.setState({ activeKey6: key })
  }

  onMenu7Change (key) {
    this.setState({ activeKey7: key })
  }

  render () {
    const { activeKey4, activeKey5, activeKey6, activeKey7 } = this.state
    return (<div className='menu-box'>
      <div className='menu-unit'>
        <h3>基础的Menu</h3>
        <Menu mode='pop' className='menu1'>
          <Menu.Item onClick={this.onItem1Click}>Edit</Menu.Item>
          <Menu.Item onClick={this.onItem1Click}>Share</Menu.Item>
          <Menu.Item onClick={this.onItem1Click}>Download</Menu.Item>
          <Menu.Item onClick={this.onItem1Click}>Delete</Menu.Item>
        </Menu>
      </div>
      
      <div className='menu-unit'>
        <h3>带Icon的Menu</h3>
        <Menu mode='pop' className='menu2'>
          <Menu.Item><Icon type='edit' />Edit</Menu.Item>
          <Menu.Item><Icon type='share1' />Share</Menu.Item>
          <Menu.Item><Icon type='download' />Download</Menu.Item>
          <Menu.Item><Icon type='delete' />Delete</Menu.Item>
        </Menu>
      </div>

      <div className='menu-unit'>
        <h3>自定义Item样式的Menu</h3>
        <Menu className='menu3'>
          <Menu.Item className='item3'>Claim Your Podcast</Menu.Item>
          <Menu.Item className='item3'>Discover</Menu.Item>
          <Menu.Item className='item3'>Sign Up</Menu.Item>
        </Menu>
      </div>

      <div className='menu-unit'>
        <h3>activeKey的Menu</h3>
        <Menu 
          mode='inline' 
          activeKey={activeKey4}
          onChange={this.onMenu4Change}
          className='menu4'>
          <Menu.Item key='dashboard' className='item4'>Dashboard</Menu.Item>
          <Menu.Item key='income' className='item4'>Income Center</Menu.Item>
          <Menu.Item key='episodes' className='item4'>Episodes</Menu.Item>
          <Menu.Item key='setting' className='item4'>Show Settings</Menu.Item>
        </Menu>
      </div>

      <div className='menu-unit'>
        <h3>mode=inline的Menu</h3>
        <Menu 
          mode='inline' 
          activeKey={activeKey5}
          onChange={this.onMenu5Change}
          className='menu5'>
          <Menu.Item key='dashboard' className='item5'>Dashboard</Menu.Item>
          <Menu.SubMenu
            title={<span>Income Center</span>}>
            <Menu.Item key='income1' className='item5'>Income 1</Menu.Item>
            <Menu.Item key='income2' className='item5'>Income 2</Menu.Item>
          </Menu.SubMenu>
          <Menu.Item key='episodes' className='item5'>Episodes</Menu.Item>
          <Menu.Item key='setting' className='item5'>Show Settings</Menu.Item>
          <Menu.Item key='analytics' className='item5'>Analytics</Menu.Item>
        </Menu>
      </div>

      <div className='menu-unit'>
        <h3>mode=pop的Menu</h3>
        <Menu 
          mode='pop' 
          activeKey={activeKey6}
          onChange={this.onMenu6Change}
          className='menu6'>
          <Menu.Item key='dashboard' className='item6'>Dashboard</Menu.Item>
          <Menu.SubMenu
            title={<span>Income Center</span>}>
            <Menu.Item key='income1' className='item6'>Income 1</Menu.Item>
            <Menu.Item key='income2' className='item6'>Income 2</Menu.Item>
            <Menu.SubMenu
              title={<span>Center</span>}>
              <Menu.Item key='center1' className='item6'>Center 1</Menu.Item>
              <Menu.Item key='center2' className='item6'>Center 2</Menu.Item>
              <Menu.Item key='center3' className='item6'>Center 3</Menu.Item>
              <Menu.Item key='center4' className='item6'>Center 4</Menu.Item>
            </Menu.SubMenu>
            <Menu.Item key='income3' className='item6'>Income 3</Menu.Item>
            <Menu.Item key='income4' className='item6'>Income 4</Menu.Item>
          </Menu.SubMenu>
          <Menu.Item key='episodes' className='item6'>Episodes</Menu.Item>
          <Menu.Item key='setting' className='item6'>Show Settings</Menu.Item>
          <Menu.Item key='analytics' className='item6'>Analytics</Menu.Item>
        </Menu>
      </div>

      <div className='menu-unit'>
        <h3>mode=replace的Menu</h3>
        <Menu 
          mode='replace' 
          activeKey={activeKey7}
          onChange={this.onMenu7Change}
          className='menu7'>
          <Menu.Item key='dashboard' className='item7'>Dashboard</Menu.Item>
          <Menu.SubMenu
            title={<span>Income Center</span>}>
            <Menu.Item key='income1' className='item7'>Income 1</Menu.Item>
            <Menu.Item key='income2' className='item7'>Income 2</Menu.Item>
          </Menu.SubMenu>
          <Menu.Item key='episodes' className='item7'>Episodes</Menu.Item>
          <Menu.Item key='setting' className='item7'>Show Settings</Menu.Item>
          <Menu.Item key='analytics' className='item7'>Analytics</Menu.Item>
        </Menu>
      </div>

      <style>
        {`
          .menu-box .menu-unit {
            display: inline-block;
            width: 25%;
          }
          .menu-box .menu1 {
            width: 200px;
            border: 1px solid #eee;
          }
          .menu-box .menu2 {
            width: 200px;
            border: 1px solid #eee;
          }
          .menu-box .menu3 {
            width: 200px;
            border: 1px solid #eee;
          }
          .menu-box .menu3 .item3 {
            height: 55px;
            line-height: 55px;
          }
          .menu-box .menu3 .item3:hover {
            color: red;
          }
          .menu-box .menu4 .item4.hmly-menu-item-checked::before {
            position: absolute;
            top: 0px;
            left: 0px;
            content: '';
            width: 4px;
            height: 40px;
            background-color: #ff4444;
          }
          .menu-box .menu5 .item5.hmly-menu-item-checked::before {
            position: absolute;
            top: 0px;
            left: 0px;
            content: '';
            width: 4px;
            height: 40px;
            background-color: #ff4444;
          }
          .menu-box .menu6 .item6.hmly-menu-item-checked::before {
            position: absolute;
            top: 0px;
            left: 0px;
            content: '';
            width: 4px;
            height: 40px;
            background-color: #ffba0f;
          }
          .menu-box .menu7 .item7.hmly-menu-item-checked::before {
            position: absolute;
            top: 0px;
            left: 0px;
            content: '';
            width: 4px;
            height: 40px;
            background-color: rgba(83,118,130,0.8);
          }
          .menu-box .menu4 .item4:hover {
            background: #fff;
          }
          .menu-box .hmly-icon {
            margin-right: 10px;
          }
        `}
      </style>
    </div>
    )
  }
}

<Example />
```
