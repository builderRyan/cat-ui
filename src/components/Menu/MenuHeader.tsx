import React, { FC } from "react";



const MenuHeader: FC = (porps)=> {
  return <div className="cat-menu-header">
    <div className="cat-menu-header-container">{porps.children}</div><i/></div>;
}

export default MenuHeader;