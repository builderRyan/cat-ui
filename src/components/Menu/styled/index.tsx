import styled from 'styled-components'
import { primaryBtnColor } from '../../../styles/theme'
import { InlineSubMenuProps, PopSubMenuBoxProps } from '../interface'

const StyledMenu = styled.ul`
  list-style: none;
  padding: 0;
`;

const StyledMenuItem = styled.li`
  color: #4a4a4a;
  width: 100%;
  height: 40px;
  padding: 0 22px;
  line-height: 40px;
  font-size: 14px;
  font-weight: 400;
  font-style: normal;
  font-stretch: normal;
  letter-spacing: normal;
  text-align: left;
  box-sizing: border-box;
  position: relative;
  cursor: pointer;
  white-space:nowrap;
  &:hover {
    background-color: #f3f3f3;
    color: #4a4a4a;
    font-weight: 600;
  }
  &.hmly-menu-item-checked {
    font-weight: bold;
    color: #3a3a3a;
  }
`;

const StyledSubMenu = styled.li`
  height: auto;
  position: relative;
`;

const SubMenuTitle = styled.div`
  color: #4a4a4a;
  width: 100%;
  height: 40px;
  padding: 0 22px;
  line-height: 40px;
  font-size: 14px;
  font-weight: 400;
  font-style: normal;
  font-stretch: normal;
  letter-spacing: normal;
  text-align: left;
  box-sizing: border-box;
  position: relative;
  cursor: pointer;
  &:hover {
    background-color: #f3f3f3;
    color: #4a4a4a;
    font-weight: 600;
  }
  &.hmly-submenu-title-inline::after {
    position: absolute;
    top: calc(50% - 8px);
    right: 25px;
    height: 10px;
    width: 10px;
    content: " ";
    border-right: 2px solid #9B9B9B;
    border-top: 2px solid #9B9B9B;
    border-left: 2px solid transparent;
    border-bottom: 2px solid transparent;
    display: inline-block;
    transform: rotate(135deg);
    box-sizing: border-box;
  }
  &.hmly-submenu-title-pop::after,
  &.hmly-submenu-title-replace::after {
    position: absolute;
    top: calc(50% - 5px);
    right: 10px;
    height: 10px;
    width: 10px;
    content: " ";
    border-right: 2px solid #9B9B9B;
    border-top: 2px solid #9B9B9B;
    border-left: 2px solid transparent;
    border-bottom: 2px solid transparent;
    display: inline-block;
    transform: rotate(45deg);
    box-sizing: border-box;
  }
  &.hmly-submenu-title-inline.hmly-submenu-title-expand::after {
    transform: rotate(-45deg);
  }
`;

const InlineSubMenu = styled(StyledMenu)`
  height: ${(props: InlineSubMenuProps) => props.height};
  overflow: hidden;
  padding-left: 20px;
`;

const PopSubMenuBox = styled.div`
  position: absolute;
  top: 0;
  right: ${(props: PopSubMenuBoxProps) => props.right};
  z-index: 999;
  width: ${(props: PopSubMenuBoxProps) => props.width};
  border: 1px solid #eee;
  box-sizing: border-box;
`

const PopSubMenu = styled.ul`
  list-style: none;
  padding: 0;
  background-color: #fff;
  box-sizing: border-box;
`

const BackToMenu = styled.div`
  color: #4a4a4a;
  width: 100%;
  height: 40px;
  padding: 0 22px 0 45px;
  line-height: 40px;
  font-size: 14px;
  font-weight: 400;
  font-style: normal;
  font-stretch: normal;
  position: relative;
  cursor: pointer;
  &::after {
    position: absolute;
    top: calc(50% - 5px);
    left: 25px;
    height: 10px;
    width: 10px;
    content: " ";
    border-right: 2px solid #9B9B9B;
    border-top: 2px solid #9B9B9B;
    border-left: 2px solid transparent;
    border-bottom: 2px solid transparent;
    display: inline-block;
    transform: rotate(225deg);
    box-sizing: border-box;
  }
`

const StyledAnimateHeight = styled.div``;

export { StyledMenu, StyledMenuItem, StyledSubMenu, SubMenuTitle,
  InlineSubMenu, PopSubMenu, PopSubMenuBox, StyledAnimateHeight,
  BackToMenu }
