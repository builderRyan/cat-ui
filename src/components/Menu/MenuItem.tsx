import * as React from 'react'
import { Component } from 'react'
import classNames from 'classnames'
import { StyledMenuItem } from './styled'

export interface MenuItemProps {
  /** 类名 */
  className?: string;
  /** 样式 */
  style?: object;
  /** 菜单类型，现在支持扩展弹出、行内内嵌、原地替换三种 */
  mode?: 'pop' | 'inline' | 'replace';
  /** 当前MenuItem的id */
  id?: boolean;
  /** 是否选中 */
  checked?: boolean;
  /** 是否禁用 */
  disabled?: boolean;
  /** MenuItem点击事件的回调 */
  onClick?: any;
  /** Menu改变事件的回调 */
  onMenuChange?: any;
  /** 是否有data-fixed属性 */
  fixed?: boolean;
}

/**
 * MenuItem
 */
class MenuItem extends Component<MenuItemProps, any> {
  static defaultProps = {
    fixed: false,
    disabled: false
  }

  onItemClick = (e: any) => {
    const { onClick, onMenuChange, id, disabled } = this.props
    if (!disabled) {
      onClick && onClick(e)
      onMenuChange && onMenuChange(id)
    }
  }

  render() {
    const { className, style, mode, checked, fixed, disabled, children } = this.props
    const classes = classNames('hmly-menu-item', {
      [`hmly-menu-item-${mode}`]: mode,
      [`hmly-menu-item-checked`]: checked,
      [`hmly-menu-item-disabled`]: disabled
    }, className)
    // console.log('MenuItem', mode, children)

    return (
      <StyledMenuItem
        data-fixed={fixed}
        className={classes}
        style={style}
        onClick={this.onItemClick}>
        {children}
      </StyledMenuItem>
    )
  }
}

export default MenuItem
