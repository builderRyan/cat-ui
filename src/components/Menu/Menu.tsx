import * as React from 'react'
import { Component, cloneElement } from 'react'
import classNames from 'classnames'
import { StyledMenu } from './styled'
import MenuItem from './MenuItem'

export interface MenuProps {
  /** 类名 */
  className?: string;
  /** 样式 */
  style?: object;
  /** 菜单类型，现在支持扩展弹出、行内内嵌、原地替换三种 */
  mode?: 'pop' | 'inline' | 'replace';
  /** active的item的key */
  activeKey?: string;
  /** 点击事件的回调 */
  onChange?: any;
}

/**
 * 菜单
 */
class Menu extends Component<MenuProps, any> {
  static Item: typeof MenuItem

  static defaultProps = {
    mode: 'pop'
  }

  onMenuChange = (id: any) => {
    // console.log('Menu:onMenuChange', id)
    const { onChange } = this.props
    onChange && onChange(id)
  }

  render() {
    const self = this
    const { className, style, mode, activeKey, children} = this.props
    const classes = classNames('hmly-menu', {
      [`hmly-menu-${mode}`]: mode
    }, className)

    const items = React.Children.map(children, (element: any, index) => {
      if (!element) { return element }
      return cloneElement(element, {
        key: index,
        id: element.key,
        checked: activeKey === element.key,
        onMenuChange: self.onMenuChange
      })
    })

    return (
      <StyledMenu
        className={classes}
        style={style}>
        {items}
      </StyledMenu>
    )
  }
}

export default Menu
