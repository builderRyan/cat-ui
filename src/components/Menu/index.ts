import Menu from './Menu'
import MenuItem from './MenuItem'
import SubMenu from './SubMenu'
import MenuHeader from './MenuHeader';

Menu.SubMenu = SubMenu
Menu.Item = MenuItem
Menu.Header = MenuHeader;

export default Menu
