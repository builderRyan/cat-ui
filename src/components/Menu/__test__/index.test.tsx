import * as React from 'react'
import * as Enzyme from 'enzyme'
import { render, mount } from 'enzyme'
import * as Adapter from 'enzyme-adapter-react-16'
import Menu from '../index'

Enzyme.configure({ adapter: new Adapter() })

describe('Menu', () => {
  test('renders correctly', () => {
    const wrapper = render(
      <Menu mode='pop' className='menu1'>
        <Menu.Item>Edit</Menu.Item>
        <Menu.Item>Share</Menu.Item>
        <Menu.Item>Download</Menu.Item>
        <Menu.Item>Delete</Menu.Item>
      </Menu>
    )
    expect(wrapper).toMatchSnapshot()
  })

})
