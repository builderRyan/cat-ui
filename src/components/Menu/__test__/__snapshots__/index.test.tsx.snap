// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu renders correctly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "hmly-menu hmly-menu-pop menu1 sc-bdVaJa dSMDdb",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "hmly-menu-item hmly-menu-item-pop sc-bwzfXH bfHbC",
          "data-fixed": "false",
        },
        "children": Array [
          Object {
            "data": "Edit",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "li",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "hmly-menu-item hmly-menu-item-pop sc-bwzfXH bfHbC",
            "data-fixed": "false",
          },
          "children": Array [
            Object {
              "data": "Share",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "li",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": Object {
            "attribs": Object {
              "class": "hmly-menu-item hmly-menu-item-pop sc-bwzfXH bfHbC",
              "data-fixed": "false",
            },
            "children": Array [
              Object {
                "data": "Download",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "li",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "class": "hmly-menu-item hmly-menu-item-pop sc-bwzfXH bfHbC",
                "data-fixed": "false",
              },
              "children": Array [
                Object {
                  "data": "Delete",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "li",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "data-fixed": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "data-fixed": undefined,
              },
            },
            "parent": [Circular],
            "prev": [Circular],
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "data-fixed": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "data-fixed": undefined,
            },
          },
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "data-fixed": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "data-fixed": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "data-fixed": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "data-fixed": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "hmly-menu-item hmly-menu-item-pop sc-bwzfXH bfHbC",
          "data-fixed": "false",
        },
        "children": Array [
          Object {
            "data": "Share",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "li",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "hmly-menu-item hmly-menu-item-pop sc-bwzfXH bfHbC",
            "data-fixed": "false",
          },
          "children": Array [
            Object {
              "data": "Download",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "li",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": Object {
            "attribs": Object {
              "class": "hmly-menu-item hmly-menu-item-pop sc-bwzfXH bfHbC",
              "data-fixed": "false",
            },
            "children": Array [
              Object {
                "data": "Delete",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "li",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": [Circular],
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "data-fixed": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "data-fixed": undefined,
            },
          },
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "data-fixed": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "data-fixed": undefined,
          },
        },
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "hmly-menu-item hmly-menu-item-pop sc-bwzfXH bfHbC",
            "data-fixed": "false",
          },
          "children": Array [
            Object {
              "data": "Edit",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "li",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "data-fixed": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "data-fixed": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "data-fixed": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "data-fixed": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "hmly-menu-item hmly-menu-item-pop sc-bwzfXH bfHbC",
          "data-fixed": "false",
        },
        "children": Array [
          Object {
            "data": "Download",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "li",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "hmly-menu-item hmly-menu-item-pop sc-bwzfXH bfHbC",
            "data-fixed": "false",
          },
          "children": Array [
            Object {
              "data": "Delete",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "li",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "data-fixed": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "data-fixed": undefined,
          },
        },
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "hmly-menu-item hmly-menu-item-pop sc-bwzfXH bfHbC",
            "data-fixed": "false",
          },
          "children": Array [
            Object {
              "data": "Share",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "li",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": Object {
            "attribs": Object {
              "class": "hmly-menu-item hmly-menu-item-pop sc-bwzfXH bfHbC",
              "data-fixed": "false",
            },
            "children": Array [
              Object {
                "data": "Edit",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "li",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": [Circular],
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "data-fixed": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "data-fixed": undefined,
            },
          },
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "data-fixed": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "data-fixed": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "data-fixed": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "data-fixed": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "hmly-menu-item hmly-menu-item-pop sc-bwzfXH bfHbC",
          "data-fixed": "false",
        },
        "children": Array [
          Object {
            "data": "Delete",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "li",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "hmly-menu-item hmly-menu-item-pop sc-bwzfXH bfHbC",
            "data-fixed": "false",
          },
          "children": Array [
            Object {
              "data": "Download",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "li",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": Object {
            "attribs": Object {
              "class": "hmly-menu-item hmly-menu-item-pop sc-bwzfXH bfHbC",
              "data-fixed": "false",
            },
            "children": Array [
              Object {
                "data": "Share",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "li",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": [Circular],
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "class": "hmly-menu-item hmly-menu-item-pop sc-bwzfXH bfHbC",
                "data-fixed": "false",
              },
              "children": Array [
                Object {
                  "data": "Edit",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "li",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "data-fixed": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "data-fixed": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "data-fixed": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "data-fixed": undefined,
            },
          },
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "data-fixed": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "data-fixed": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "data-fixed": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "data-fixed": undefined,
        },
      },
    ],
    "name": "ul",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
